### PR TITLE
Update s6-overlay to v3.1.0.1 for 3.15

### PR DESCRIPTION
**Pull request by nemchik**
* s6 v3.x changes the structure of the init, so we can no longer patch stage2 the way we were. Instead we are patching into the `rc.init` to run `/docker-mods` before any other init script.
* v3.x has also done away with the installer, so this switches back to adding and unpacking the tar files.
* v3.x also no longer ships `/usr/bin/with-contenv` but instead offers separately installable symlinks (which I've added to the Dockerfile), one of which links `/usr/bin/with-contenv` to `/command/with-contenv` so can directly replace `/usr/bin/with-contenv` without renaming it to `/usr/bin/with-contenvb` and then within our replacement we can call `/command/with-contenv`
* All s6 stuff (adding, unpacking, patching) is moved before the final layer.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -63,6 +63,9 @@ ARG VERSION
 LABEL build_version="Linuxserver.io version:- ${VERSION} Build-date:- ${BUILD_DATE}"
 LABEL maintainer="TheLamer"
 
+# set version for s6 overlay
+ARG S6_OVERLAY_VERSION="3.1.0.1"
+
```
Comment (nemchik) [inline Dockerfile]: Not needed (see comment below about `ci-check`)

File: Dockerfile [hunk 2]
```diff
@@ -97,5 +100,6 @@ RUN \
 
 # add local files
 COPY root/ /
+COPY ci-check/ /package/admin/s6-overlay-${S6_OVERLAY_VERSION}/etc/s6-rc/sources/top/contents.d
```
Comment (nemchik) [inline Dockerfile]: Move this up around like 50 (where the docker mod stuff is patched in) and change it to:
```suggestion
COPY ci-check/ /root-out/package/admin/s6-overlay-${S6_OVERLAY_VERSION}/etc/s6-rc/sources/top/contents.d
```

**Comment by ksurl** [comment]
3.16 has the installer in alpine repo. I'm not sure if they intend to package for 3.15 as well

**Comment by nemchik** [comment]
> 3.16 has the installer in alpine repo. I'm not sure if they intend to package for 3.15 as well

That's actually pretty cool. We probably won't start using it until it's a little more available.

**Review by thespad — APPROVED** [review]
LGTM

**Review by Roxedus — APPROVED** [review]

**Review by aptalca — APPROVED** [review]

**Comment by dolfinus** [comment]
This change probably caused issue https://github.com/linuxserver/docker-openssh-server/issues/59 in openssh-server image, which is based on 3.15, but using s6 v2

**Pull request merged**